libs/template/config.go

```diff
@@ -99,9 +99,38 @@ func (c *config) assignDefaultValues(r *renderer) error {
 	return nil
 }
 
+func (c *config) isSkipped(p jsonschema.Property) bool {
+	if p.Schema.SkipPromptIf == nil {
+		return false
+	}
+
+	// Check if conditions for skipping are satisfied
+	skip := true
+	for name, property := range p.Schema.SkipPromptIf.Properties {
+		if v, ok := c.values[name]; ok && v == property.Const {
+			continue
+		}
+		skip = false
+		break
+	}
+
+	// Skip prompting and assign default value if conditions are satisfied
+	if skip {
+		c.values[p.Name] = p.Schema.Default
+	}
+
+	return skip
+}
+
 // Prompts user for values for properties that do not have a value set yet
 func (c *config) promptForValues(r *renderer) error {
 	for _, p := range c.schema.OrderedProperties() {
+
+		// Skip prompting if we can.
+		if c.isSkipped(p) {
```

Comment:
The function name suggests this is a function call without side effects where in fact it writes to `c.values`.

It shouldn't write to `c.values` directly but instead leverage the code below where the default value is interpreted as a template. If you move this block to under the `defaultVal` computation it should be good. Any reason not to do this?

Reply:
Done. Assigned default value after template computation now!

Reply:
Moved default value computation to a the skip function. This keeps the flow of code cleaner.